src/md/src/DevGuides/async-reporting.md

```diff
@@ -0,0 +1,166 @@
+    1. [Scheme](#scheme)
+    1. [Finishing launch](#finishing-launch)
+---
+
+### Overview
+
+Asynchronous reporting implemented using [AMQP 0-9-1](https://www.rabbitmq.com/tutorials/amqp-concepts.html) protocol based on 
+[RabbitMq](https://www.rabbitmq.com) message broker.
+The main idea of async reporting is to give response back immediately after server receiving request from client.
+So using that approach client is not blocked and don't wait until server process request.
+
+### Simple scheme of interactions between RabbitMq and API
+
+***Difference between ID and UUID***  
+
+`ID` is physical identificator of entity generated automatic by database at the moment of saving.  
+`UUID` is virtual identificator of entity. Can be specified in request or if not present in request generated automatic at the moment 
+`api` accepts request.  
+Each entity has both `ID` and `UUID`. `ID` is used to perform CRUD operation with ***already saved in db*** entity. 
+`UUID` is used to build child-parent relationships between entities at client side during reporting. 
+In case synchronous reporting any response from `api` returns ***after*** handling request and saving entity in database. 
+In case asynchronous reporting any response from `api` returns ***before*** handling request and saving entity in database.
+Responses in both modes looks the same:
+```json
+{
+  "id": "cd64d5eb-fea1-4e7e-8a5a-69998ac5620f"
+}
+```
+`id` property of response actually is `UUID`. This is due to backward compatibility. 
+So when you have this uuid and want to update, delete ent entity you should get physical `id` from db first. 
+
+***Simple asyncronous reporting scheme***  
+
+
+* **Step 1**  
+`API` receives HTTP request from `client`. `Controller` checks permissions and throws request to `producer`.
+* **Step 2**  
+`Producer` validates business rules if necessary, generates UUID(virtual id) for entity and returns it back to `controller`, 
+composes message for `RabbitMq` and 
+sends it to specified queue.
+After that `controller` returns HTTP response to `client` that contains UUID. **At the moment physical entity may not be created!**
+* **Step 3**  
+`Consumer` starts processing the message as soon as it received from `RabbitMq`. 
+After successfully processed entity will be stored in database and obtain physical id. 
+In case exception occurs it logged and entity will not be saved.
+
+![](/src/Images/devguide/async/simple-scheme.png)
+
+### Enable asynchronous reporting in agents
+
+Async reporting is supported only in agents since version 5.0.0.
+To enable it you should set `rp.reporting.async=true` (*) in `reportportal.properties`.
+By default (if property `rp.reporting.async` is not specified) agents work in synchronous mode.
+
+```properties
+rp.endpoint=https://example.com
+rp.api.key=xxx
+rp.launch=launch-name
+rp.project=project-name
+rp.reporting.async=true
+```
+
+(*) For java based agents only. Enabling in other agents may be different.
```

Comment:
(*) Listed above is an example for Java-based client. For another platforms please see corresponding documentation

Reply:
done